### ISSUE TITLE
Installation from alternative repository (test.pypi.org) fails due to missing pip option

### ISSUE THREAD

**Issue by konstin** [sdispater/poetry#173]
I have uploaded a dummy package `get_fourtytwo` on testpypi. When I install it with pip directly, it works fine:

```
pip install --index-url https://test.pypi.org/simple/ get_fourtytwo
python -c "import get_fourtytwo"
```

I tried to install the package through poetry using this pyproject.toml:

```
[tool.poetry]
name = "foo"
version = "0.1.0"
description = ""
authors = ["konstin <konstin@mailbox.org>"]

[tool.poetry.dependencies]
python = "=3.6"
get_fourtytwo = "*"

[[tool.poetry.source]]
name = "testpypisimple"
url = "https://test.pypi.org/simple/"
```

`poetry --version`: `Poetry 0.10.2`

`poetry install -vvv`:

```
Updating dependencies
Resolving dependencies...
   1: fact: foo is 0.1.0
   1: derived: foo
   1: fact: foo depends on get_fourtytwo (*)
   1: selecting foo (0.1.0)
   1: derived: get_fourtytwo (*)
   1: selecting get-fourtytwo (1.6.0)
   1: Version solving took 0.043 seconds.
   1: Tried 1 solutions.


Package operations: 1 install, 0 updates, 0 removals

Writing lock file

  - Installing get-fourtytwo (1.6.0)
                                                                                                                                                                
[VenvCommandError]                                                                                                                               
Command ['pip', 'install', '--no-deps', '-r', '/tmp/get-fourtytwo-1.6.0cvvv8zj_reqs.txt'] errored with the following output:                                  
Collecting get-fourtytwo==1.6.0 (from -r /tmp/get-fourtytwo-1.6.0cvvv8zj_reqs.txt (line 1))                                                                   
  Could not find a version that satisfies the requirement get-fourtytwo==1.6.0 (from -r /tmp/get-fourtytwo-1.6.0cvvv8zj_reqs.txt (line 1)) (from versions: )  
No matching distribution found for get-fourtytwo==1.6.0 (from -r /tmp/get-fourtytwo-1.6.0cvvv8zj_reqs.txt (line 1))                                           
                                                                                                                                                                
Exception trace:
 /usr/local/lib/python3.6/dist-packages/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /usr/local/lib/python3.6/dist-packages/poetry/console/application.py in do_run() at line 87
   return super(Application, self).do_run(i, o)
 /usr/local/lib/python3.6/dist-packages/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /usr/local/lib/python3.6/dist-packages/poetry/console/commands/command.py in run() at line 72
   return super(BaseCommand, self).run(i, o)
 /usr/local/lib/python3.6/dist-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /usr/local/lib/python3.6/dist-packages/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /usr/local/lib/python3.6/dist-packages/poetry/console/commands/install.py in handle() at line 50
   return installer.run()
 /usr/local/lib/python3.6/dist-packages/poetry/installation/installer.py in run() at line 75
   self._do_install(local_repo)
 /usr/local/lib/python3.6/dist-packages/poetry/installation/installer.py in _do_install() at line 254
   self._execute(op)
 /usr/local/lib/python3.6/dist-packages/poetry/installation/installer.py in _execute() at line 262
   getattr(self, "_execute_{}".format(method))(operation)
 /usr/local/lib/python3.6/dist-packages/poetry/installation/installer.py in _execute_install() at line 287
   self._installer.install(operation.package)
 /usr/local/lib/python3.6/dist-packages/poetry/installation/pip_installer.py in install() at line 37
   self.run(*args)
 /usr/local/lib/python3.6/dist-packages/poetry/installation/pip_installer.py in run() at line 62
   return self._venv.run("pip", *args, **kwargs)
 /usr/local/lib/python3.6/dist-packages/poetry/utils/venv.py in run() at line 274
   raise VenvCommandError(e)

install [--no-dev] [--dry-run] [-E|--extras EXTRAS] [--develop DEVELOP]
```

While the version and the sha256 in /tmp/get-fourtytwo-1.6.0cvvv8zj_reqs.txt are correct, it the installation fails because the `index-url` parameter is missing, i.e. `pip install --no-deps --index-url https://test.pypi.org/simple/ -r get-fourtytwo-1.6.0cvvv8zj_reqs.txt` works fine. 

(Note that `get_fourtytwo` is a bdist_whl only available for cp3.6m on linux and testpypi gets cleaned sometimes, so you might to upload a different package to testpypi to reproduce)

**Comment by tlecarrour** [comment]
Same problem here!
Seems that in order for the `--index-url` to be added to the pip command the package must have `source_type` set to "legacy" and a value for `source_url` (`poetry/installation/pip_installer.py:20`). It's not clear to me where those values have to be set.

**Comment by sdispater** [comment]
This should be fixed in the latest release (`0.10.3`)

**Comment by konstin** [comment]
Indeed, thank you for the quick fix! (And, of course, for the overall great work)
